runtime/gc_realtime/RealtimeRootScanner.hpp

```diff
@@ -117,5 +118,376 @@ class MM_RealtimeRootScanner : public MM_RootScanner
 	}
 };
 
+/**
+ * This scanner will mark objects that pass through its doSlot.
+ */
+class MM_RealtimeMarkingSchemeRootMarker : public MM_RealtimeRootScanner
```

Comment:
Would you please create RealtimeMarkingSchemeRootMarker.hpp (and RealtimeMarkingSchemeRootMarker.cpp if necessary) and move this code there?